v2g-liberty/rootfs/root/appdaemon/apps/v2g_liberty/reservations_client.py

```diff
@@ -225,14 +226,28 @@ async def activate_selected_calendar(self):
         self.poll_timer_id = await self.hass.run_every(
             self.__poll_calendar, "now", self.POLLING_INTERVAL_SECONDS
         )
-        # self.hass.log(f"activate_selected_calendar started polling_time {self.poll_timer_id} "
+        # self.__log(f"activate_selected_calendar started polling_time {self.poll_timer_id} "
         #          f"every {self.POLLING_INTERVAL_SECONDS} sec.")
-        self.hass.log("Completed activate_selected_calendar")
+        self.__log("Completed activate_selected_calendar")
 
     ######################################################################
     #                   PRIVATE (CALLBACK) FUNCTIONS                     #
     ######################################################################
 
+    def __log(self, msg:str, level:str=""):
+        info = inspect.stack()[1][0]
```

Comment:
Do you know how "expensive" it is to do this?

For the purpose of adding the class-name, I think we can fix that by initialising a log function in the constructor with a "partial" that captures the class-name.
But, ofcourse, then you will not have the benefit of also automatically capturing the method name.

Reply:
https://gist.github.com/JettJones/c236494013f22723c1822126df944b12 may give a good indication on what the "best" way to do this is.